registry/storage/driver/gcs/gcs.go

```diff
@@ -751,57 +750,68 @@ func (d *driver) Delete(context context.Context, path string) error {
 		}
 		return nil
 	}
-	err = storageDeleteObject(gcsContext, d.bucket, d.pathToKey(path))
-	if err != nil {
-		if status, ok := err.(*googleapi.Error); ok {
-			if status.Code == http.StatusNotFound {
-				return storagedriver.PathNotFoundError{Path: path}
-			}
-		}
+	err = storageDeleteObject(ctx, d.bucket, d.pathToKey(path), d.GCS)
+	if err == storage.ErrObjectNotExist {
+		return storagedriver.PathNotFoundError{Path: path}
 	}
 	return err
 }
 
-func storageDeleteObject(context context.Context, bucket string, name string) error {
-	return retry(func() error {
-		return storage.DeleteObject(context, bucket, name)
-	})
+func storageDeleteObject(ctx context.Context, bucket string, name string, gcs *storage.Client) error {
+	if err := gcs.Bucket(bucket).Object(name).Delete(ctx); err != nil {
+		return err
+	}
+	return nil
 }
 
-func storageStatObject(context context.Context, bucket string, name string) (*storage.Object, error) {
-	var obj *storage.Object
+func storageStatObject(ctx context.Context, bucket string, name string, gcs *storage.Client) (*storage.ObjectAttrs, error) {
+	bkt := gcs.Bucket(bucket)
+	var obj *storage.ObjectAttrs
 	err := retry(func() error {
 		var err error
-		obj, err = storage.StatObject(context, bucket, name)
+		obj, err = bkt.Object(name).Attrs(ctx)
 		return err
 	})
 	return obj, err
 }
 
-func storageListObjects(context context.Context, bucket string, q *storage.Query) (*storage.Objects, error) {
-	var objs *storage.Objects
-	err := retry(func() error {
-		var err error
-		objs, err = storage.ListObjects(context, bucket, q)
-		return err
-	})
-	return objs, err
+func storageListObjects(ctx context.Context, bucket string, q *storage.Query, gcs *storage.Client) ([]*storage.ObjectAttrs, error) {
+	bkt := gcs.Bucket(bucket)
+	objs := []*storage.ObjectAttrs{}
+	it := bkt.Objects(ctx, q)
+	for {
+		objAttrs, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, err
+		}
+		objs = append(objs, objAttrs)
+	}
+
+	return objs, nil
 }
 
-func storageCopyObject(context context.Context, srcBucket, srcName string, destBucket, destName string, attrs *storage.ObjectAttrs) (*storage.Object, error) {
-	var obj *storage.Object
-	err := retry(func() error {
-		var err error
-		obj, err = storage.CopyObject(context, srcBucket, srcName, destBucket, destName, attrs)
-		return err
-	})
-	return obj, err
+func storageCopyObject(ctx context.Context, srcBucket, srcName string, destBucket, destName string, attrs *storage.ObjectAttrs, gcs *storage.Client) (*storage.ObjectAttrs, error) {
+	src := gcs.Bucket(srcBucket).Object(srcName)
+	dst := gcs.Bucket(destBucket).Object(destName)
+	attrs, err := dst.CopierFrom(src).Run(ctx)
+	if err != nil {
+		if status, ok := err.(*googleapi.Error); ok {
+			if status.Code == http.StatusNotFound {
+				return nil, storagedriver.PathNotFoundError{Path: srcName}
+			}
+		}
+		return nil, fmt.Errorf("Object(%q).CopierFrom(%q).Run: %v", destName, srcName, err)
```

Comment:
Any reason not to wrap the error?
```suggestion
		return nil, fmt.Errorf("Object(%q).CopierFrom(%q).Run: %w", destName, srcName, err)
```